test/SpokePoolClient.ValidateFill.ts

```diff
@@ -256,62 +254,82 @@ describe("SpokePoolClient: Fill Validation", async function () {
-      await spokePoolClient1.binarySearchForBlockContainingDepositId(4)
-    );
-    expect(await spokePoolClient1.binarySearchForBlockContainingDepositId(5)).to.be.greaterThanOrEqual(
-      await spokePoolClient1.binarySearchForBlockContainingDepositId(3)
+    expect(await spokePoolClient1.getBlockRangeForDepositId(5)).to.be.greaterThanOrEqual(
+      await spokePoolClient1.getBlockRangeForDepositId(3)
     );
 
     // Last deposit ID is now stricly less than last block in search range because there were more blocks mined since
     // the deposit that incrementerd ID from 4 to 5.
-    expect(await spokePoolClient1.binarySearchForBlockContainingDepositId(5)).to.be.lessThan(
-      spokePoolClient1.latestBlockNumber
-    );
+    expect(await spokePoolClient1.getBlockRangeForDepositId(5)).to.be.lessThan(spokePoolClient1.latestBlockNumber);
 
-    expect(await spokePoolClient1.binarySearchForBlockContainingDepositId(2)).to.be.greaterThanOrEqual(
-      depositEvents[1].blockNumber
-    );
-    expect(await spokePoolClient1.binarySearchForBlockContainingDepositId(2)).to.be.lessThan(
-      depositEvents[3].blockNumber
+    expect(await spokePoolClient1.getBlockRangeForDepositId(2)).to.be.greaterThanOrEqual(depositEvents[1].blockNumber);
+    expect(await spokePoolClient1.getBlockRangeForDepositId(2)).to.be.lessThan(depositEvents[3].blockNumber);
+  });
+
+  it("Fuzz: get search bounds for deposit ID", async function () {
+    const fuzzClient = new MockedSpokePoolClient(
+      createSpyLogger().spyLogger,
+      spokePool_2,
+      null,
+      destinationChainId,
+      spokePool2DeploymentBlock
     );
+
+    const initLow = 0;
+    const initHigh = 1000000;
+    const depositIds = Array(initHigh - initLow + 1).fill(0);
+
+    const testIterations = 100;
+    for (let i = 0; i < testIterations; i++) {
+      // Randomize deposit ID's between initLow and initHigh. ID's should only increase
+      // and will do so 50% of the time. The other 50% of the time they will stay the same.
+      for (let j = 1; j < depositIds.length; j++) {
+        const increment = Math.random() > 0.5 ? 1 : 0;
+        depositIds[j] = depositIds[j - 1] + increment;
+      }
+      fuzzClient.setDepositIds(depositIds);
+
+      // Randomize target between highest and lowest values in deposit IDs.
+      const target = Math.floor(Math.random() * (depositIds[depositIds.length - 1] - initLow)) + initLow;
+
+      // Randomize max # of searches.
+      const maxSearches = Math.floor(Math.random() * 19) + 1;
+      const results = await fuzzClient._getBlockRangeForDepositId(target, initLow, initHigh, maxSearches);
+
+      expect(results.low >= initLow).to.be.true;
+      expect(results.high <= initHigh).to.be.true;
+      expect(results.mid <= results.high && results.mid >= results.low).to.be.true;
+      expect(depositIds[results.low] <= target).to.be.true;
+      expect(depositIds[results.high] > target).to.be.true;
```

Comment:
Can we add one more check to actually find the exact block and make sure it's in those bounds?
```suggestion
      expect(depositIds[results.high] > target).to.be.true;
      // The correct block is the first block to increment beyond the target.
      const correctBlock = depositIds.find(depositId => depositId > target);
      expect(correctBlock <= results.high).to.be.true;
      expect(results.low <= correctBlock).to.be.true;
```

Reply:
This is a good catch, have refactored the tests a bit